numpy/polynomial/chebyshev.py

```diff
@@ -1594,6 +1700,70 @@ def chebvander3d(x, y, z, deg):
     return v.reshape(v.shape[:-3] + (-1,))
 
 
+def chebvandernd(deg, *xi):
+    """Pseudo-Vandermonde matrix of given degrees.
+
+    Returns the pseudo-Vandermonde matrix of degrees `deg` and sample
+    points `(x1, x2, ..., xn)`. If `l, m, n` are the given degrees in
+    `x1, x2, x3`, then The pseudo-Vandermonde matrix is defined by
+
+    .. math:: V[..., (m+1)(n+1)i + (n+1)j + k] = T_i(x1)*T_j(x2)*T_k(x3),
```

Comment:
This is not correct for the general nd case. Might be tricky to express it for the general case.